src/OptionsProvider/OptionsProvider/ServiceCollectionExtensions.cs

```diff
@@ -19,11 +20,12 @@ public static class ServiceCollectionExtensions
 		string path)
 	{
 		services.AddMemoryCache();
-		services.AddSingleton<IOptionsLoader, OptionsLoader>();
-		services.AddSingleton(serviceProvider =>
+		services.AddTransient<IOptionsProviderBuilder, OptionsProviderBuilder>();
```

Comment:
Maybe we shouldn't bind it directly and we could use ` ActivatorUtilities.CreateInstance` below